packages/connect-common/src/messageChannel/abstract.ts

```diff
@@ -72,10 +72,8 @@ export abstract class AbstractMessageChannel<
      * initiates handshake sequence with peer. resolves after communication with peer is established
      */
     public init() {
-        if (!this.handshakeFinished) {
```

Comment:
I am trying to recall why it was here. do we call somewher init multiple times? is it even allowed, or is it expected usecase? if not, we should maybe return some error here?

Reply:
I am not particulary happy with removing this. It means that now it would be possible to trigger `handshakeWithPeer` multiple times which imho might result into some unwanted racecondition.

Reply:
Yes, this is probably not the right place to do this.
Maybe we could clear the messageChannel so it initialize fresh when we need it.

For example when we clear popup https://github.com/trezor/trezor-suite/blob/feat/connect-explorer-serviceworker-proxy/packages/connect-webextension/src/popup.ts#L145 we are doing `this.channel.disconnect();` if we are disconnecting the channel I guess we should clear it in a way that new connection is is possible.

Reply:
What about something like https://github.com/trezor/trezor-suite/pull/10426/commits/aac9cce1bdec2a5a979b5d73672fad00b400c888 ?

Reply:
that sounds reasonable. 👍